src/main/java/frc/robot/subsystems/CorallatorSubsystem.java

```diff
@@ -1,51 +1,75 @@
 package frc.robot.subsystems;
 
+import org.opencv.video.SparseOpticalFlow;
+
 import com.revrobotics.RelativeEncoder;
 import com.revrobotics.spark.SparkAnalogSensor;
+import com.revrobotics.spark.SparkBase.PersistMode;
+import com.revrobotics.spark.SparkBase.ResetMode;
 import com.revrobotics.spark.SparkFlex;
+import com.revrobotics.spark.SparkLimitSwitch;
 import com.revrobotics.spark.SparkLowLevel.MotorType;
 import com.revrobotics.spark.SparkMax;
 
 import edu.wpi.first.math.controller.PIDController;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Configs;
 
-public class CorallatorSubsystem extends SubsystemBase{
+public class CorallatorSubsystem extends SubsystemBase {
     private SparkMax m_wrist = new SparkMax(12, MotorType.kBrushless);
     private RelativeEncoder m_wristEncoder = m_wrist.getEncoder();
-    private SparkFlex m_flinger = new SparkFlex(13, MotorType.kBrushless);
-    private final PIDController m_pidController = new PIDController(.02, 0, 0);
-    private final double targetAngleUpDegrees = 8;
-    private final double targetAngleDownDegrees = -11;
-    private double set_point;
-    public double flingerSpeed = 0.15; 
-     
-    public CorallatorSubsystem(){
-        // set_point = m_wristEncoder.getPosition();
-        set_point = targetAngleDownDegrees;
+    private SparkFlex m_corallator = new SparkFlex(13, MotorType.kBrushless);
+    private SparkLimitSwitch m_limitSwitch = m_corallator.getForwardLimitSwitch();
+    private final PIDController m_pidController = new PIDController(.04, 0, 0);
+    private final double targetAngleUpDegrees = 14; // FIND THE RIGHT ANGLES!!!!!!! :3
+    private final double targetAngleDownDegrees = -7.5;
+    private double setPoint;
+    public double flingerSpeed = 0.05;
+
+    public CorallatorSubsystem() {
+        m_wristEncoder.setPosition(0);
```

Comment:
It appears that we have some duplicated lines here.  Please delete lines 31 and 32.   The `setPoint` variable should probably be set to zero so the Corallator won't move when we start the code.